code/modules/mob/living/carbon/life.dm

```diff
@@ -39,6 +39,10 @@
 	//Updates the number of stored chemicals for powers
 	handle_changeling()
 
+	if(mind)
+		for(var/key in mind.addiction_points)
+			var/datum/addiction/addiction = SSaddiction.all_addictions[key]
+			addiction.process_addiction(src)
```

Comment:
Can you code ahead, and pass a dummy deltatime into this proc, like so.
```suggestion
			addiction.process_addiction(src, delta_time = 2)
```

And then have that delta time used for all the `prob()` calls of the addiction. Then when Life delta_time gets merged, it'll be easy to integrate with this or vice versa.

Reply:
pain but ok